app/views/user.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404
+
+from ..models import User
+from ..serializers import UserSerializer
+
+
+class UserList(APIView):
+
+    def get(self, request):
+        users = User.objects.all()
+        serializer = UserSerializer(users, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = UserSerializer(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class UserDetail(APIView):
+
+    def get_object(self, pk):
+        try:
+            return User.objects.get(user_id=pk)
+        except User.DoesNotExist:
+            raise Http404
```

Comment:
I think you should return a response here instead of raising an error.  `return Response(serializer.errors, status=status.HTTP_404_NOT_FOUND)`